includes/utilities.inc

```diff
@@ -205,3 +205,70 @@ function islandora_batch_scan_check_encoding($type, $encoding) {
     }
   }
 }
+
+/**
+ * Batch to delete objects ingested via a batch set.
+ *
+ * @param int $set_id
+ *   Batch set ID.
+ */
+function islandora_batch_batch_delete_ingested_object_in_set($set_id) {
+  module_load_include('inc', 'islandora_batch', 'includes/db');
+  $batch = array(
+    'title' => t('Delete ingested objects from set'),
+    'finished' => 'islandora_batch_batch_delete_ingested_object_in_set_finished',
+    'init_message' => t('Initializing...'),
+    'operations' => array(),
+    'progress_message' => t('Time elapsed: @elapsed.'),
+    'file' => drupal_get_path('module', 'islandora_batch') . '/includes/utilities.inc',
+  );
+
+  $items = islandora_batch_get_queued_items_for_set($set_id);
```

Comment:
Not sure where to comment, but we should probably acquire the lock for the set (when doing every operation here involving objects in the set, really)...

May be easier if moving to the who single-op, multiple-iteration bit we usually use:
```php
[... the previous stuff from here ...]
  $batch['operations'][] = array('islandora_batch_batch_delete_ingested_objects_in_set_operation', array($set_id));
  batch_set($batch);
}

function islandora_batch_batch_delete_ingested_objects_in_set_operation($set_id, &$context) {
  module_load_include('inc', 'islandora_batch', 'includes/ingest.batch');
  $sandbox =& $context['sandbox'];
  if (!islandora_batch_get_lock($set_id)) {
    // Failed to acquire the lock... Try again in another iteration...
    $context['finished'] = 0;
    return;
  }

  if (!isset($sandbox['total'])) {
    $sandbox['items'] = islandora_batch_get_queued_items_for_set($set_id);
    $sandbox['total'] = count($sandbox['items']);
    $sandbox['progress'] = 0;
    if ($sandbox['total'] === 0) {
      $context['message'] = t('Nothing to process...');
      return;
    }
  }
  // Might be increased to avoid bootstrapping overhead.
  $slice_size = 10;
  $slice_stop = $sandbox['progress'] + $slice_size;

  for (; $sandbox['progress'] < $slice_stop; $sandbox['progress']++) {
    $item_id = $sandbox['items'][$sandbox['progress']];
    $object = islandora_object_load($item_id);
    if ($object) {
      if ($object->repository->purgeObject($object->id)) {
        $context['results']['purged'][] = $item_id;
      }
      else {
        $context['results']['purged_fail'][] = $item_id;
      }
    }
  }
  $context['finished'] = $sandbox['progress'] / $sandbox['total'];
}
```